src/main/java/platform/codingnomads/co/springweb/resttemplate/GET/getForObject/GetForObjectDemo.java

```diff
@@ -24,13 +28,35 @@ public static void main(String[] args) {
     @Bean
     public CommandLineRunner run() throws Exception {
         return args -> {
-
+            // Quotes
             QuoteTemplate[] randomQuote;
             randomQuote = restTemplate.getForObject("https://zenquotes.io/api/random/", QuoteTemplate[].class);
             System.out.println(Arrays.toString(randomQuote));
 
+            // Excuses
+            ExcuseTemplate[] randomExcuse;
+            randomExcuse = restTemplate.getForObject("https://excuser-three.vercel.app/v1/excuse/party/2", ExcuseTemplate[].class);
+            System.out.println(Arrays.toString(randomExcuse));
+
+            // Bored
+            BoredTemplate randomActivity;
+            randomActivity = restTemplate.getForObject("http://www.boredapi.com/api/activity?participants=1", BoredTemplate.class);
+            System.out.println(randomActivity);
+
+            // Pass parameters with Map
+            HashMap<String, Integer> boredApiParameters = new HashMap<String, Integer>();
+            boredApiParameters.put("participants", 2);
+//            randomActivity = restTemplate.getForObject("http://www.boredapi.com/api/activity?participants=2", BoredTemplate.class, boredApiParameters);
+//            System.out.println(randomActivity);
+            ResponseEntity<BoredTemplate> responseEntity = restTemplate.getForEntity("http://www.boredapi.com/api/activity?participants=2", BoredTemplate.class, boredApiParameters);
```

Comment:
The `boredApiParameters` map values will not be used in this scenario. To fix you need to use `{ ... }` named parameter: 
```suggestion
            ResponseEntity<BoredTemplate> responseEntity = restTemplate.getForEntity("http://www.boredapi.com/api/activity?participants={participants}", BoredTemplate.class, boredApiParameters);
```
There are many ways to build the url, I'll show you several ways on 1-1.

Reply:
Yes! Ok, thanks for catching that.